File: signature/jquery.go
package signature

var JQuerySignature = Signature{
	Name:        "jQuery",
	Description: "A fast, small, and feature-rich JavaScript library.",
	Cpe:         "cpe:/a:jquery:jquery",

	DetectPattern: DetectPattern{
		URLs: []string{
			"jquery",
		},
		Bodies: []string{
			"jquery",
			"jQuery",
		},
	},
	VersionPattern: VersionPattern{
		URLs: []string{
			`jquery[@/-](\d+\.\d+\.\d+)`,
		},
		Bodies: []string{
			`jQuery v(\d+\.\d+\.\d+)`,
			`jQuery JavaScript Library v(\d+\.\d+\.\d+)`,
			`var C="(\d+\.\d+\.\d+)"`,
			`d="(\d+\.\d+\.\d+)"`,
		},
	},
}

File: src/signatures/technologies/jquery.ts
import type { Signature } from "../_types.js";

export const jquerySignature: Signature = {
  name: "jQuery",
  description:
    "jQuery is a JavaScript library which is a free, open-source software designed to simplify HTML DOM tree traversal and manipulation, as well as event handling, CSS animation, and Ajax.",
  cpe: "cpe:/a:jquery:jquery",
  rule: {
    confidence: "high",
    urls: [
      "/jquery(?:-(\\d+\\.\\d+\\.\\d+))?(?:\\.slim)?(?:\\.min)?\\.js",
      "(\\d+\\.\\d+\\.\\d+)/jquery(?:\\.slim)?(?:\\.min)?\\.js",
    ],
    javascriptVariables: {
      "$.fn.jquery": "(\\d+\\.\\d+\\.\\d+)",
      "jQuery.fn.jquery": "(\\d+\\.\\d+\\.\\d+)",
    },
  },
};
